src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1138,3 +1138,280 @@ def list_container_services(client, resource_group_name=None):
+        public_ip_address_id = '{}/publicIPAddresses/{}'.format(network_id_template, public_ip_address) if public_ip_address else None  # pylint: disable=line-too-long
+
+        # calculate default names if not provided
+        backend_pool_name = backend_pool_name or '{}BEPool'.format(load_balancer)
+        nat_pool_name = nat_pool_name or '{}NatPool'.format(load_balancer)
+        if not backend_port:
+            backend_port = 3389 if os_type == 'windows' else 22
+
+        master_template.add_resource(build_load_balancer_inbound_nat_rules_resource(  # pylint: disable=too-many-function-args
+            load_balancer, location, tags, backend_port, instance_count, 'loadBalancerFrontEnd'))
+
+        lb_resource = build_load_balancer_resource(
+            load_balancer, location, tags, backend_pool_name, nat_pool_name, backend_port,
+            'loadBalancerFrontEnd', public_ip_address_id, subnet_id)
+        lb_resource['dependsOn'] = lb_dependencies
+        master_template.add_resource(lb_resource)
+
+    if storage_profile in [StorageProfile.SACustomImage, StorageProfile.SAPirImage]:
+        master_template.add_resource(build_vmss_storage_account_pool_resource(
+            'storageLoop', location, tags, storage_sku))
+        vmss_dependencies.append('storageLoop')
+    else:
+        raise CLIError('Unsupported storage profile.')
+
+    naming_prefix = '{}{}'.format(vmss_name[:5], random_string(4, force_lower=True))
+    backend_address_pool_id = '{}/loadBalancers/{}/backendAddressPools/{}'.format(
+        network_id_template, load_balancer, backend_pool_name) if load_balancer_type else None
+    inbound_nat_pool_id = '{}/loadBalancers/{}/inboundNatPools/{}'.format(
+        network_id_template, load_balancer, nat_pool_name) if load_balancer_type == 'new' else None
+    ip_config_name = '{}IPConfig'.format(naming_prefix)
+    nic_name = '{}Nic'.format(naming_prefix)
+
+    vmss_resource = build_vmss_resource(vmss_name, naming_prefix, location, tags,
+                                        not disable_overprovision, upgrade_policy_mode,
+                                        vm_sku, instance_count,
+                                        ip_config_name, nic_name, subnet_id, admin_username,
+                                        authentication_type, storage_profile,
+                                        os_disk_name, storage_caching, os_type,
+                                        image, admin_password,
+                                        ssh_key_value, ssh_dest_key_path,
+                                        os_publisher, os_offer, os_sku, os_version,
+                                        backend_address_pool_id, inbound_nat_pool_id)
+    vmss_resource['dependsOn'] = vmss_dependencies
+
+    master_template.add_resource(vmss_resource)
+    master_template.add_variable('storageAccountNames', [
+        '{}{}'.format(naming_prefix, x) for x in range(5)
```

Comment:
make sure we get rid of `-` in the naming_prefix which is not a valid name for a storage account